src/main/java/io/cdap/plugin/dlp/Utils.java

```diff
@@ -0,0 +1,217 @@
+            break;
+
+          case TIMESTAMP_MICROS:
+          case TIMESTAMP_MILLIS:
+            Timestamp timestampValue = fieldValue.getTimestampValue();
+            ZoneId zoneId = oldRecord.getTimestamp(fieldName).getZone();
+            LocalDateTime localDateTime;
+            if (timestampValue.getSeconds() + timestampValue.getNanos() == 0) {
+              localDateTime = LocalDateTime
+                .parse(fieldValue.getStringValue(), DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm'Z'"));
+
+            } else {
+              localDateTime = Instant
+                .ofEpochSecond(timestampValue.getSeconds(), timestampValue.getNanos())
+                .atZone(zoneId)
+                .toLocalDateTime();
+            }
+
+            recordBuilder.setTimestamp(fieldName, ZonedDateTime.of(localDateTime, zoneId));
+            break;
+          case DATE:
+            Date dateValue = fieldValue.getDateValue();
+            recordBuilder
+              .setDate(fieldName, LocalDate.of(dateValue.getYear(), dateValue.getMonth(), dateValue.getDay()));
+            break;
+          default:
+            throw new IllegalArgumentException("Failed to parse table into structured record");
+
+        }
+
+
+      } else {
+        recordBuilder.convertAndSet(fieldName, fieldValue.getStringValue());
+      }
+    }
+    return recordBuilder.build();
+  }
+
+  public static StructuredRecord.Builder createBuilderFromStructuredRecord(StructuredRecord record) {
+    StructuredRecord.Builder recordBuilder = StructuredRecord.builder(record.getSchema());
+    for (Schema.Field field : record.getSchema().getFields()) {
+      String fieldName = field.getName();
+      Object fieldValue = record.get(fieldName);
+      Schema fieldSchema = field.getSchema().isNullable() ? field.getSchema().getNonNullable() : field.getSchema();
+
+      final Schema.LogicalType logicalType = fieldSchema.getLogicalType();
+      if (fieldSchema.getType().isSimpleType()) {
+        recordBuilder.set(fieldName, fieldValue);
+      } else {
+        if (logicalType != null) {
+          switch (logicalType) {
+            case TIME_MICROS:
+            case TIME_MILLIS:
+              recordBuilder.setTime(fieldName, (LocalTime) fieldValue);
+              break;
+            case TIMESTAMP_MICROS:
+            case TIMESTAMP_MILLIS:
+              recordBuilder.setTimestamp(fieldName, (ZonedDateTime) fieldValue);
+              break;
+            case DATE:
+              recordBuilder.setDate(fieldName, (LocalDate) fieldValue);
+              break;
+            default:
+              throw new IllegalArgumentException(
+                String
+                  .format("DLP plugin does not support type '%s' for field '%s'", logicalType.toString(), fieldName));
+
+          }
+        }
+      }
+    }
+    return recordBuilder;
+  }
+
+  public static Table getTableFromStructuredRecord(StructuredRecord record) throws Exception {
```

Comment:
nothing throws exception so we can remove `throws Exception`

Reply:
There is a Throw in both of the switch blocks, should I change it to `throws IllegalStateException`?